pkg/util/test/util.go

```diff
@@ -2331,3 +2387,28 @@ func CreateClusterWithTLS(caCertFileName, serverCertFileName, serverKeyFileName
 func BoolPtr(val bool) *bool {
 	return &val
 }
+
+// NewK8sClient returns a new controller runtime Kubernetes client
+func NewK8sClient(scheme *runtime.Scheme) (client.Client, error) {
```

Comment:
due to loop import issue I dup the whole function here, don't know if there are better solutions, please let me know @piyush-nimbalkar

Reply:
One way we can avoid this is by using the [sched-ops function - GetNodes](https://github.com/portworx/sched-ops/blob/master/k8s/core/nodes.go#L22). It is fine to directly talk to the k8s API directly in tests.

If we do that then we can split `GetTopologySpreadConstraints` and extract the portion of getting nodes. The test could use GetNodes and pass the node list to the function calculating the spread constraints.

```
// Actual code will call this
func GetTopologySpreadConstraints(k8sClient, labels) {
  nodeList := ...
  k8sClient.Get(... nodeList ...)
  GetTopologySpreadConstraintsForNodes(nodeList, labels)
}

// Above test will call this
func GetTopologySpreadConstraintsForNodes(nodeList, labels) {}
```